lib/grape_logging/middleware/request_logger.rb

```diff
@@ -23,34 +27,71 @@ def initialize(app, options = {})
       def before
         reset_db_runtime
         start_time
-
         invoke_included_loggers(:before)
       end
 
-      def after
+      def after(status, response)
         stop_time
+
+        # Response status
+        @response_status = status
+        @response_body   = response
+
+        # Perform repotters
         @reporter.perform(collect_parameters)
+
+        # Invoke loggers
         invoke_included_loggers(:after)
         nil
       end
 
+      # Call stack and parse responses & status.
+      #
+      # @note Exceptions are logged as 500 status & re-raised.
       def call!(env)
-        super
-      end
+        @env = env
 
-      protected
+        # Before hook
+        before
 
-      def response
-        begin
-          super
-        rescue
+        # Catch error
+        error = catch(:error) do
+          begin
+            @app_response = @app.call(@env)
+          rescue => e
+            # Log as 500 + message
+            after(500, e.message)
```

Comment:
Hi @Strnadj, this is not correct. It always logs 500, even if the actual error code is different. For example, for `Grape::Exceptions::MethodNotAllowed` it logs:

```
{"status":500,"time":{"total":39.52,"db":3.81,"view":35.71},"method":"DELETE","path":"/papi/ping","params":{},"host":"localhost"}
```

It should be:
```
{"status":405,"time":{"total":39.52,"db":3.81,"view":35.71},"method":"DELETE","path":"/papi/ping","params":{},"host":"localhost"}
```

Change to `after(e.status, e.message)` fixes the issue.